scheduler/replica_scheduler.go

```diff
@@ -574,42 +600,23 @@ func (rcs *ReplicaScheduler) RequireNewReplica(replicas map[string]*longhorn.Rep
 		return 0
 	}
 
-	// Otherwise Longhorn will relay the new replica creation then there is a chance to reuse failed replicas later.
-	settingValue, err := rcs.ds.GetSettingAsInt(types.SettingNameReplicaReplenishmentWaitInterval)
+	timeUntilNext, timeOfNext, err := rcs.timeToReplacementReplica(volume)
 	if err != nil {
-		logrus.Errorf("Failed to get Setting ReplicaReplenishmentWaitInterval, will directly replenish a new replica: %v", err)
-		return 0
+		msg := "Failed to get time until replica replacement, will directly replenish a new replica"
+		logrus.WithError(err).Errorf(msg)
```

Comment:
Yes, this part is a bit awkward. It's the compromise I ended up with, because:

- I wanted to reuse code between `getDiskCandidates` and `RequireNewReplica`. 
- `getDiskCandidates` would not like us to log anything every time it runs, so we need to log at the call site in `RequireNewReplica`.
- We need to parse `volume.Status.LastDegradedAt` in the shared code to calculate `timeUntilNext`, so I didn't want to also parse it in RequireNewReplica to calculate `timeOfNext`.

Reply:
Further addressed in latest commit by:

- Remove possibility of negative return from `timeToReplacementReplica`.
- Add the skew second in `RequireNewReplica` instead of `timeToReplacementReplica` in case a caller doesn't want it.

I don't think the underestimation matters. There's no guarantee we reconcile again at the time mentioned anyway (maybe the controller will be busy, etc.). It's just a helpful log message.